src/newrelic/collector/client.go

```diff
@@ -215,12 +215,16 @@ type clientImpl struct {
 	httpClient *http.Client
 }
 
-func (c *clientImpl) perform(url string, data []byte, userAgent string, requestHeadersMap map[string]string) ([]byte, error) {
+func (c *clientImpl) perform(url string, data []byte, userAgent string, requestHeadersMap map[string]string, maxPayloadSize int) ([]byte, error) {
 	deflated, err := Compress(data)
 	if nil != err {
 		return nil, err
 	}
 
+	if l := deflated.Len(); l > maxPayloadSize {
```

Comment:
If the payload exceeds the maximum size we should also update the Supportability metric `Supportability/Agent/Collector/MaxPayloadSizeLimit/<endpoint>` as outlined [here](https://source.datanerd.us/agents/agent-specs/blob/68735b2884ccc836fb2a148a4814bd420bdeb647/Connect-LEGACY.md#max-payload-size).